### PR TITLE
feat: User defined GitOps Bug Fixes & Concurrent cases handling

### PR DESCRIPTION
**Pull request by dhananjay0601**
# Description

-> Global configuration for the setting up the repository type for the user
-> Adding Drawer to for the helm apps so that user can configure the Repository type for application deployment.
-> Setting up the GitOps validation and adding repository type for the devtron app

Fixes https://github.com/devtron-labs/devtron/issues/4308

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have performed a self-review of my own code
* [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appConfig/appConfig.scss [hunk 1]
```diff
@@ -82,6 +82,75 @@
             }
         }
     }
+    .app-compose-with-gitops-config__nav {
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/appConfig.scss:85]: Please minimise the duplicacy

File: src/components/app/details/appConfig/AppComposeRouter.tsx [hunk 2]
```diff
@@ -124,6 +127,20 @@ export default function AppComposeRouter({
...
+        if (isUnlocked.gitOpsConfig) {
+            return URLS.APP_GITOPS_CONFIG
+        } else if (isUnlocked.deploymentTemplate) {
+            return URLS.APP_DEPLOYMENT_CONFIG
+        } else if (isUnlocked.dockerBuildConfig) {
+            return URLS.APP_DOCKER_CONFIG
+        } else if (isUnlocked.workflowEditor) {
+            return URLS.APP_WORKFLOW_CONFIG
+        } else {
+            return URLS.APP_GITOPS_CONFIG
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/AppComposeRouter.tsx:140]: It should be `APP_GIT_CONFIG`. Please verify and fix

File: src/components/app/details/appConfig/AppConfig.tsx [hunk 3]
```diff
@@ -472,9 +480,16 @@ function Navigation({
 }: AppConfigNavigationProps) {
     const location = useLocation()
     const selectedNav = navItems.filter((navItem) => location.pathname.indexOf(navItem.href) >= 0)[0]
+    const isGitOpsConfigurationRequired = navItems.find((item) => item.stage === STAGE_NAME.GITOPS_CONFIG)?.required
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/AppConfig.tsx:483]: this should be coming from parent as we are calculating this there

File: src/components/app/details/appConfig/AppConfig.utils.ts [hunk 4]
```diff
@@ -3,59 +3,78 @@ import { DOCUMENTATION } from '../../../../config'
...
-        workflowEditor:
+        gitOpsConfig:
             stage === STAGE_NAME.CI_PIPELINE ||
             stage === STAGE_NAME.DEPLOYMENT_TEMPLATE ||
+            stage === STAGE_NAME.GITOPS_CONFIG ||
+            stage === STAGE_NAME.CD_PIPELINE ||
+            stage === STAGE_NAME.CHART_ENV_CONFIG,
+        workflowEditor:
+            stage === STAGE_NAME.CI_PIPELINE ||
+            (isGitOpsConfigurationRequired
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/AppConfig.utils.ts:40]: we can common out this to reuse

File: src/components/app/details/triggerView/workflow/nodes/triggerPrePostCDNode.tsx [hunk 5]
```diff
@@ -41,39 +52,95 @@ export class TriggerPrePostCDNode extends Component<TriggerPrePostCDNodeProps>{
...
-                        <button className="workflow-node__deploy-btn" data-testid={`${this.props.type}-trigger-select-image-${this.props.index}`} onClick={(event) => { event.stopPropagation(); context.onClickCDMaterial(this.props.id, this.props.type) }}>Select Image</button>
-                    </div>
-                </div>
-            }}
-        </TriggerViewContext.Consumer>
+        let status = this.props.status ? this.props.status.toLocaleLowerCase() : ''
+        let stage = this.props.type === 'PRECD' ? 'Pre-deployment' : 'Post-deployment'
+        let isClickable = !(
+            status === DEFAULT_STATUS.toLowerCase() ||
+            status === 'not triggered' ||
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/workflow/nodes/triggerPrePostCDNode.tsx:81]: Please use `BUILD_STATUS` constant

File: src/components/app/details/appConfig/AppConfig.utils.ts (hunk 4 above)
Comment (dhananjay0601) [reply src/components/app/details/appConfig/AppConfig.utils.ts:40]: Used const isCommonUnlocked variable to reduce duplicity

File: src/components/app/details/appConfig/AppConfig.tsx (hunk 3 above)
Comment (dhananjay0601) [reply src/components/app/details/appConfig/AppConfig.tsx:483]: Resolved

File: src/components/app/details/appConfig/AppComposeRouter.tsx (hunk 2 above)
Comment (dhananjay0601) [reply src/components/app/details/appConfig/AppComposeRouter.tsx:140]: Verified & Resolved

File: src/components/app/details/triggerView/workflow/nodes/triggerPrePostCDNode.tsx (hunk 5 above)
Comment (dhananjay0601) [reply src/components/app/details/triggerView/workflow/nodes/triggerPrePostCDNode.tsx:81]: Resolved

File: src/components/charts/modal/DeployChart.tsx [hunk 6]
```diff
@@ -234,6 +234,7 @@ const DeployChart: React.FC<DeployChartProps> = ({
                     valuesOverride: obj,
                     valuesOverrideYaml: textRef,
                     appName,
+                    gitRepoURL: 'Default',
```
Comment (vivek-devtron) [inline src/components/charts/modal/DeployChart.tsx:237]: Why is this required?

File: src/components/gitOps/GitOpsConfiguration.tsx [hunk 7]
```diff
@@ -352,23 +361,26 @@ class GitOpsConfiguration extends Component<GitOpsProps, GitOpsState> {
...
-                            validationStatus: VALIDATION_STATUS.FAILURE,
+                            validationStatus: !resp.validationSkipped ? VALIDATION_STATUS.FAILURE : '', 
                             saveLoading: false,
                             isFormEdited: false,
                             validationError: errorMap || [],
                             deleteRepoError: resp.deleteRepoFailed,
+                            validationSkipped: resp.validationSkipped,
                         })
-                        toast.error('Configuration validation failed')
+                        {this.state.selectedRepoType === repoType.DEFAULT && toast.error('Configuration failed')}
```
Comment (vivek-devtron) [inline src/components/gitOps/GitOpsConfiguration.tsx:382]: we should add condition in message

Comment (vivek-devtron) [inline src/components/gitOps/GitOpsConfiguration.tsx:382]: We should add if else here

File: src/components/gitOps/UserGitRepConfiguration.tsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+import { UserGitRepoConfigurationProps } from './gitops.type'
+import { repoType } from '../../config'
+import { ReactComponent as Warn } from '../../assets/icons/ic-warning.svg'
+import { STAGE_NAME } from '../app/details/appConfig/appConfig.type'
+import { useHistory } from 'react-router-dom'
+import { toast } from 'react-toastify'
+import { URLS } from '../../config'
+import { ReloadNoGitOpsRepoConfiguredModal } from '../workflowEditor/NoGitOpsRepoConfiguredWarning'
+
+export default function UserGitRepConfiguration(props: UserGitRepoConfigurationProps) {
```
Comment (vivek-devtron) [inline src/components/gitOps/UserGitRepConfiguration.tsx:14]: Spread the props

File: src/components/gitOps/UserGitRepConfiguration.tsx [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+        )
+    }
+
+    const closePopup = () => {
+        setShowReloadModal(false)
+    }
+
+    const reload = () => {
+        history.push(`/app/${props.appId}/edit/${URLS.APP_WORKFLOW_CONFIG}`)
+        window.location.reload()
```
Comment (vivek-devtron) [inline src/components/gitOps/UserGitRepConfiguration.tsx:87]: Why are we doing page reload here?

File: src/components/gitOps/UserGitRepo.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
+import { InfoColourBar, RadioGroup, RadioGroupItem } from '@devtron-labs/devtron-fe-common-lib'
+import React from 'react'
+import { repoType } from '../../config/constants'
+import { ReactComponent as Error } from '../../assets/icons/ic-error-exclamation.svg'
+import { ReactComponent as Warn } from '../../assets/icons/ic-warning.svg'
+import { UserGitRepoProps } from './gitops.type';
+
+function UserGitRepo(props: UserGitRepoProps) {
```
Comment (vivek-devtron) [inline src/components/gitOps/UserGitRepo.tsx:8]: Spread the props

File: src/components/gitOps/UserGitRepo.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+                <div className="ml-26 mt-8">
+                    <div className="gitops__id fw-5 fs-13 mb-8 dc__required-field">Git Repo URL</div>
+                    <input
+                        type="text"
+                        autoComplete="off"
+                        name="name"
+                        value={props.repoURL.trim()}
+                        placeholder="Enter repository URL"
+                        className="form__input"
+                        onChange={(event) => onChange(event)}
```
Comment (vivek-devtron) [inline src/components/gitOps/UserGitRepo.tsx:30]: we can write this as `onChange={OnChange}`

File: src/components/gitOps/UserGitRepConfiguration.tsx (hunk 9 above)
Comment (dhananjay0601) [reply src/components/gitOps/UserGitRepConfiguration.tsx:87]: As per the requirement if a case of stale data occurs than we have to reload the page.

File: src/components/v2/values/chartValuesDiff/ChartValuesView.type.ts [hunk 12]
```diff
@@ -90,15 +90,41 @@ export interface DeploymentAppSelectorType {
     handleDeploymentAppTypeSelection?: (event) => void
     isDeployChartView: boolean
     allowedDeploymentTypes?: DeploymentAppTypes[]
+    allowedCustomBool: boolean
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.type.ts:93]: Please remove unused props

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 13]
```diff
@@ -605,7 +613,7 @@ function ChartValuesView({
                 // ends
 
                 // helm app OR external helm app delete initiated
-                if (response.result.deleteResponse?.deleteInitiated || (isExternalApp && !commonState.installedAppInfo)) {
+                if (response.result.deleteResponse?.deleteInitiated || (isExternalApp && !commonState.installedApInfo)) {
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:616]: is it intentional? `installedAppInfo` to `installedApInfo`

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 14]
```diff
@@ -882,8 +890,14 @@ function ChartValuesView({
                     valuesOverrideYaml: commonState.modifiedValuesYaml,
                     appName: appName.trim(),
                     deploymentAppType: isVirtualEnvironmentOnSelector ? DeploymentAppTypes.MANIFEST_DOWNLOAD : commonState.deploymentAppType,
-                }
+                    gitRepoURL: (staleData || commonState.gitRepoURL?.gitRepoURL === undefined)
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:893]: Do we need to check the case for null and ''?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 15]
```diff
@@ -926,6 +940,7 @@ function ChartValuesView({
                 const {
                     result: { environmentId: newEnvironmentId, installedAppId: newInstalledAppId },
                 } = res
+                console.log(isDeployChartView)
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:943]: Please remove this console.log

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 16]
```diff
@@ -1545,16 +1567,48 @@ function ChartValuesView({
...
+                            !appDetails?.isVirtualEnvironment &&
+                            !commonState.installedConfig?.isOCICompliantChart && (
                                 <DeploymentAppSelector
                                     commonState={commonState}
                                     isUpdate={isUpdate}
                                     handleDeploymentAppTypeSelection={handleDeploymentAppTypeSelection}
                                     isDeployChartView={isDeployChartView}
                                     allowedDeploymentTypes={allowedDeploymentTypes}
+                                    allowedCustomBool={allowedCustomBool}
+                                    envId={commonState.selectedEnvironment ? commonState.selectedEnvironment.value : 0}
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:1579]: Please remove the unnecessary props.

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 17]
```diff
@@ -1545,16 +1567,48 @@ function ChartValuesView({
...
+                                />
+                            )}
+                        {!isExternalApp &&
+                            !isCreateValueView &&
+                            !isVirtualEnvironmentOnSelector &&
+                            (isDeployChartView || allowedDeploymentTypes.length > 0) &&
+                            !appDetails?.isVirtualEnvironment &&
+                            !commonState.installedConfig?.isOCICompliantChart && (
+                                <GitOpsDrawer
+                                    commonState={commonState}
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:1596]: same here

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 18]
```diff
@@ -204,51 +225,197 @@ const RadioWithTippy = (children, isFromCDPipeline: boolean, tippyContent: strin
...
+                </div>
+            )}
+            {(gitOpsState && allowedDeploymentTypes.indexOf(DeploymentAppTypes.HELM) !== -1) || (showRepoSelector && window._env_.HIDE_GITOPS_OR_HELM_OPTION) ? (
+                <div className="form__input dashed mt-10 flex" style={{ height: '54px' }}>
+                    <div className="mb-10">
+                        <span>
+                            Commit deployment manifests to
+                            <EditIcon className="icon-dim-16 cursor ml-28 pt-4" onClick={toggleDrawer} />
+                        </span>
+                        <a className="dc__ellipsis-right flex left fs-13 fw-4 lh-20 cursor pb-4" onClick={toggleDrawer}>{`${
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:411]: Please simplify this nested ternary to increase readability

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 16 above)
Comment (dhananjay0601) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx]: Resolved

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 17 above)
Comment (dhananjay0601) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx]: Resolved

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx (hunk 18 above)
Comment (dhananjay0601) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx]: Resolved

File: src/components/cdPipeline/NewCDPipeline.tsx [hunk 19]
```diff
@@ -51,7 +51,19 @@ import { pipelineContext } from '../workflowEditor/workflowEditor'
...
-import { DeleteDialogType, ForceDeleteMessageType, NewCDPipelineProps } from './types'
+import NoGitOpsRepoConfiguredWarning from '../workflowEditor/NoGitOpsRepoConfiguredWarning'
+import {
+    gitOpsRepoNotConfigured,
+    gitOpsRepoNotConfiguredWithEnforcedEnv,
+    gitOpsRepoNotConfiguredWithOptionsHidden,
+} from '../gitOps/constants'
+import { NewCDPipelineProps, DeleteDialogType, ForceDeleteMessageType } from './types'
+
+export enum deleteDialogType {
```
[the remark this answers is not in the document]
Comment (dhananjay0601) [reply src/components/cdPipeline/NewCDPipeline.tsx]: Moved

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

File: src/components/app/details/appConfig/AppComposeRouter.tsx [hunk 20]
```diff
@@ -211,6 +227,7 @@ export default function AppComposeRouter({
                         )}
                     />,
                 ]}
+                <Redirect to={`${maximumAllowedUrl}`} />
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/AppComposeRouter.tsx:230]: Can we write it as to={maximumAllowedUrl}. Also can we rename this?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.scss [hunk 21]
```diff
@@ -446,3 +446,19 @@
         }
     }
 }
+
+.repo-url-link {
+    display: block;
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.scss:451]: Can we use the formulae classes?

File: src/components/cdPipeline/NewCDPipeline.tsx [hunk 22]
```diff
@@ -61,19 +67,28 @@ export default function NewCDPipeline({
     getWorkflows,
     refreshParentWorkflows,
     envIds,
+    isLastNode,
```
Comment (vivek-devtron) [inline src/components/cdPipeline/NewCDPipeline.tsx:70]: Please remove this if not required

File: src/components/gitOps/UserGitRepConfiguration.tsx [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+import { toast } from 'react-toastify'
+import { gitOpsConfigDevtron, getGitOpsRepoConfig } from '../../services/service'
+import UserGitRepo from './UserGitRepo'
+import { UserGitRepoConfigurationProps } from './gitops.type'
+import { repoType } from '../../config'
+import { ReactComponent as Warn } from '../../assets/icons/ic-warning.svg'
+import { STAGE_NAME } from '../app/details/appConfig/appConfig.type'
+import { ReloadNoGitOpsRepoConfiguredModal } from '../workflowEditor/NoGitOpsRepoConfiguredWarning'
+
+const UserGitRepConfiguration: FunctionComponent<UserGitRepoConfigurationProps> = ({
```
Comment (vivek-devtron) [inline src/components/gitOps/UserGitRepConfiguration.tsx:13]: why are we using this syntax?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 24]
```diff
@@ -205,51 +237,219 @@ const RadioWithTippy = (children, isFromCDPipeline: boolean, tippyContent: strin
...
+                        value={DeploymentAppTypes.GITOPS}
+                        disabled={allowedDeploymentTypes.indexOf(DeploymentAppTypes.GITOPS) === -1}
+                    >
+                        GitOps
+                    </RadioGroupItem>
+                </ConditionalWrap>
+            </RadioGroup>
+            {deploymentAppType === DeploymentAppTypes.GITOPS &&
+                isGitOpsRepoNotConfigured &&
+                !window._env_.HIDE_GITOPS_OR_HELM_OPTION && (
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:299]: we can remove this `!window._env_.HIDE_GITOPS_OR_HELM_OPTION` as this component will only load in that case

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 25]
```diff
@@ -205,51 +237,219 @@ const RadioWithTippy = (children, isFromCDPipeline: boolean, tippyContent: strin
...
+        setGitOpsState(true)
+        setIsDrawerOpen(false)
+        if (selectedRepoType !== repoType.CONFIGURE) {
+            setSelectedRepoType(repoType.DEFAULT)
+        }
+    }
+
+    const handleRepoTextChange = (newRepoText: string) => {
+        setRepoURL(newRepoText)
+        dispatch({
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:346]: this should happen at the time of save.

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 26]
```diff
@@ -1545,16 +1577,36 @@ function ChartValuesView({
...
                                     commonState={commonState}
                                     isUpdate={isUpdate}
                                     handleDeploymentAppTypeSelection={handleDeploymentAppTypeSelection}
                                     isDeployChartView={isDeployChartView}
                                     allowedDeploymentTypes={allowedDeploymentTypes}
+                                    gitRepoURL={installedConfigFromParent['gitRepoURL']}
+                                />
+                            )}
+                        {allowedCustomBool && !isExternalApp &&
+                            !isCreateValueView &&
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:1592]: Can we create a variable and pull out the common conditions in that?

File: src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx [hunk 27]
```diff
@@ -0,0 +1,115 @@
...
+import { ReactComponent as RetryIcon } from '../../assets/icons/ic-arrow-clockwise.svg'
+import { ButtonWithLoader } from '../common'
+import { getGitOpsRepoConfig } from '../../services/service'
+
+export const ReloadNoGitOpsRepoConfiguredModal: FunctionComponent<ReloadNoGitOpsRepoConfiguredModalType> = ({
+    closePopup,
+    reload,
+}: ReloadNoGitOpsRepoConfiguredModalType) => {
+    return (
+        <VisibleModal2 className="confirmation-dialog">
```
Comment (vivek-devtron) [inline src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx:18]: Any specific reason to choose `VisibleModal2`?

File: src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx [hunk 28]
```diff
@@ -0,0 +1,115 @@
...
+    }
+    return (
+        <>
+            {reloadNoGitOpsRepoConfiguredModal ? (
+                <ReloadNoGitOpsRepoConfiguredModal
+                    closePopup={closeReloadNoGitOpsRepoConfiguredModal}
+                    reload={reload}
+                />
+            ) : (
+                <VisibleModal2 className="confirmation-dialog">
```
Comment (vivek-devtron) [inline src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx:81]: Same here

File: src/components/v2/values/chartValuesDiff/ChartValuesView.scss (hunk 21 above)
Comment (mukultayal135) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.scss:451]: done

File: src/components/app/details/appConfig/AppComposeRouter.tsx (hunk 20 above)
Comment (mukultayal135) [reply src/components/app/details/appConfig/AppComposeRouter.tsx:230]: done

File: src/components/cdPipeline/NewCDPipeline.tsx (hunk 22 above)
Comment (mukultayal135) [reply src/components/cdPipeline/NewCDPipeline.tsx:70]: done

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx (hunk 24 above)
Comment (mukultayal135) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:299]: yes, done

File: src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx (hunk 27 above)
Comment (mukultayal135) [reply src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx:18]: was already there, have changed to VisibleModal

File: src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx (hunk 28 above)
Comment (mukultayal135) [reply src/components/workflowEditor/NoGitOpsRepoConfiguredWarning.tsx:81]: done

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 29]
```diff
@@ -205,51 +237,219 @@ const RadioWithTippy = (children, isFromCDPipeline: boolean, tippyContent: strin
...
+    const [repoURL, setRepoURL] = useState('')
+
+    useEffect(() => {
+        if (deploymentAppType === DeploymentAppTypes.GITOPS) {
+            setIsDeploymentAllowed(allowedDeploymentTypes.indexOf(DeploymentAppTypes.GITOPS) !== -1)
+            setRepoURL('')
+        } else {
+            setGitOpsState(false)
+        }
+    }, [deploymentAppType, allowedDeploymentTypes, window._env_.HIDE_GITOPS_OR_HELM_OPTION])
```
[the remark this answers is not in the document]
Comment (mukultayal135) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:329]: I didn't work on this component, this seems to be redundant here

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx (hunk 25 above)
Comment (mukultayal135) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:346]: done

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 30]
```diff
@@ -890,6 +898,12 @@ const ChartValuesView = ({
                     deploymentAppType: isVirtualEnvironmentOnSelector
                         ? DeploymentAppTypes.MANIFEST_DOWNLOAD
                         : commonState.deploymentAppType,
+                    gitRepoURL:
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:901]: Please simplify this condition to avoid nested ternary. Use const for 'Default'

Comment (mukultayal135) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:901]: Have simplified logic in refactor branch

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]
